Warnings with clang and '-Wconversion'

Using clang++ version 19.1.7 on Red Hat:

```
> unzip pugixml-1.15.zip
> cd pugixml-1.15/Misc
> clang++ -c "-Wconversion" pugixml.cpp -o pugixml.o
pugixml.cpp:694:92: warning: implicit conversion changes signedness: 'unsigned long' to 'ptrdiff_t' (aka 'long') [-Wsign-conversion]
  694 |                         ptrdiff_t page_offset = reinterpret_cast<char*>(header) - reinterpret_cast<char*>(page) - sizeof(xml_memory_page);
      |                                   ~~~~~~~~~~~   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
pugixml.cpp:694:60: warning: implicit conversion changes signedness: 'long' to 'unsigned long' [-Wsign-conversion]
  694 |                         ptrdiff_t page_offset = reinterpret_cast<char*>(header) - reinterpret_cast<char*>(page) - sizeof(xml_memory_page);
      |                                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ~
pugixml.cpp:696:11: warning: implicit conversion changes signedness: 'ptrdiff_t' (aka 'long') to 'uintptr_t' (aka 'unsigned long') [-Wsign-conversion]
  696 |                         assert(page_offset % xml_memory_block_alignment == 0);
      |                                ^~~~~~~~~~~ ~

...

pugixml.cpp:4745:62: warning: implicit conversion changes signedness: 'long' to 'size_t' (aka 'unsigned long') [-Wsign-conversion]
 4745 |                 return strcpy_insitu(dest, header, header_mask, begin, end - begin);
      |                        ~~~~~~~~~~~~~                                   ~~~~^~~~~~~
pugixml.cpp:5570:16: note: in instantiation of function template specialization 'pugi::impl::(anonymous namespace)::set_value_integer<unsigned long long, char *, unsigned long>' requested here
 5570 |                 return impl::set_value_integer<unsigned long long>(_attr->value, _attr->header, impl::xml_memory_page_value_allocated_mask, rhs, rhs < 0);
      |                              ^
60 warnings generated.
```


We're using pugixml in github.com/LLNL/gidiplus, and one of our users is pushing us to fix -Wconversion warnings.

--- a/src/pugixml.cpp
+++ b/src/pugixml.cpp
@@ -54,6 +54,7 @@
 #if defined(__clang__)
 #	pragma clang diagnostic push
 #	pragma clang diagnostic ignored "-Wzero-as-null-pointer-constant" // NULL as null pointer constant
+#	pragma clang diagnostic ignored "-Wsign-conversion" // implicit conversion changes signedness
 #endif
 
 #if defined(_MSC_VER) && defined(__c2__)

